Lib9c/Action/RankingBattle8.cs

```diff
@@ -0,0 +1,287 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Numerics;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Nekoyume.Battle;
+using Nekoyume.Model.BattleStatus;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using Serilog;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionObsolete(BlockChain.Policy.BlockPolicySource.V100089ObsoleteIndex)]
+    [ActionType("ranking_battle8")]
+    public class RankingBattle8 : GameAction
+    {
+        public const int StageId = 999999;
+        public static readonly BigInteger EntranceFee = 100;
+
+        public Address avatarAddress;
+        public Address enemyAddress;
+        public Address weeklyArenaAddress;
+        public List<Guid> costumeIds;
+        public List<Guid> equipmentIds;
+        public List<Guid> consumableIds;
+        public BattleLog Result { get; private set; }
+        public AvatarState EnemyAvatarState;
+        public ArenaInfo ArenaInfo;
+        public ArenaInfo EnemyArenaInfo;
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            IActionContext ctx = context;
+            var states = ctx.PreviousStates;
+            var inventoryAddress = avatarAddress.Derive(LegacyInventoryKey);
+            var worldInformationAddress = avatarAddress.Derive(LegacyWorldInformationKey);
+            var questListAddress = avatarAddress.Derive(LegacyQuestListKey);
+            if (ctx.Rehearsal)
+            {
+                return states
+                    .SetState(avatarAddress, MarkChanged)
+                    .SetState(weeklyArenaAddress, MarkChanged)
+                    .SetState(inventoryAddress, MarkChanged)
+                    .SetState(worldInformationAddress, MarkChanged)
+                    .SetState(questListAddress, MarkChanged);
+            }
+
```

Comment:
```suggestion
CheckObsolete(BlockChain.Policy.BlockPolicySource.V100089ObsoleteIndex, context);
```
you need to add that code